### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by sLisnychyi**

### CODE REVIEW

File: gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package juja.microservices.gamification.jersey;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import ua.com.juja.gamification.model.entity.User;
```
Comment (vkuchyn) [inline gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java]: @sLisnychyi here import ua.com.... but I've found class User in package juja.microservices.gamification.model.entity
This leads to compilation error. Not only here - there lots of places listed in stacktrace below
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserController.java:[3,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserController.java:[11,4] error: cannot find symbol
[ERROR]  interface UserController
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserController.java:[13,9] error: cannot find symbol
[ERROR]  interface UserController
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[5,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[6,39] error: package ua.com.juja.gamification.service does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[17,12] error: cannot find symbol
[ERROR]  class UserControllerImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[31,11] error: cannot find symbol
[ERROR]  class UserControllerImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[39,16] error: cannot find symbol
[ERROR]  class UserControllerImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/Service.java:[3,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/Service.java:[11,4] error: cannot find symbol
[ERROR]  interface Service
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/Service.java:[13,9] error: cannot find symbol
[ERROR]  interface Service
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[5,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[6,48] error: package ua.com.juja.gamification.model.repository does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[14,12] error: cannot find symbol
[ERROR]  class ServiceImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[22,11] error: cannot find symbol
[ERROR]  class ServiceImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[27,16] error: cannot find symbol
[ERROR]  class ServiceImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryCustom.java:[3,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryCustom.java:[11,4] error: cannot find symbol
[ERROR]  interface UserRepositoryCustom
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryCustom.java:[13,9] error: cannot find symbol
[ERROR]  interface UserRepositoryCustom
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[7,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[24,11] error: cannot find symbol
[ERROR]  class UserRepositoryImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[30,16] error: cannot find symbol
[ERROR]  class UserRepositoryImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[5,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[9,56] error: cannot find symbol
[ERROR]  class User
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[13,4] error: cannot find symbol
[ERROR]  interface UserRepository
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[15,9] error: cannot find symbol
[ERROR]  interface UserRepository
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[18,8] error: cannot find symbol
[ERROR]  class UserRepositoryImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[18,24] error: cannot find symbol
[ERROR]  class UserRepositoryImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[26,44] error: cannot find symbol
[ERROR]  class UserRepositoryImpl
/Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[31,37] error: cannot find symbol
[INFO] 30 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.434s
[INFO] Finished at: Tue Apr 05 23:16:50 EEST 2016
[INFO] Final Memory: 16M/309M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:compile (default-compile) on project microservices-gamification: Compilation failure: Compilation failure:
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserController.java:[3,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserController.java:[11,4] error: cannot find symbol
[ERROR] interface UserController
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserController.java:[13,9] error: cannot find symbol
[ERROR] interface UserController
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[5,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[6,39] error: package ua.com.juja.gamification.service does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[17,12] error: cannot find symbol
[ERROR] class UserControllerImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[31,11] error: cannot find symbol
[ERROR] class UserControllerImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java:[39,16] error: cannot find symbol
[ERROR] class UserControllerImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/Service.java:[3,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/Service.java:[11,4] error: cannot find symbol
[ERROR] interface Service
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/Service.java:[13,9] error: cannot find symbol
[ERROR] interface Service
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[5,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[6,48] error: package ua.com.juja.gamification.model.repository does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[14,12] error: cannot find symbol
[ERROR] class ServiceImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[22,11] error: cannot find symbol
[ERROR] class ServiceImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/service/ServiceImpl.java:[27,16] error: cannot find symbol
[ERROR] class ServiceImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryCustom.java:[3,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryCustom.java:[11,4] error: cannot find symbol
[ERROR] interface UserRepositoryCustom
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryCustom.java:[13,9] error: cannot find symbol
[ERROR] interface UserRepositoryCustom
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[7,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[24,11] error: cannot find symbol
[ERROR] class UserRepositoryImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[30,16] error: cannot find symbol
[ERROR] class UserRepositoryImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[5,44] error: package ua.com.juja.gamification.model.entity does not exist
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[9,56] error: cannot find symbol
[ERROR] class User
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[13,4] error: cannot find symbol
[ERROR] interface UserRepository
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepository.java:[15,9] error: cannot find symbol
[ERROR] interface UserRepository
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[18,8] error: cannot find symbol
[ERROR] class UserRepositoryImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[18,24] error: cannot find symbol
[ERROR] class UserRepositoryImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[26,44] error: cannot find symbol
[ERROR] class UserRepositoryImpl
[ERROR] /Users/viktor/Projects/Personal/Juja/sandbox/microservices/gamification/src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java:[31,37] error: cannot find symbol
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

Process finished with exit code 1

```

**Comment by vkuchyn** [comment]
@sLisnychyi Thanks for your job! There are some enhancements:
1. See my comment above about compilation error
2. There are no tests. It would be great to have unit tests at least (also, perfect - integration)
3. After fixing compilation errors - please, run build with profile qulice `mvn clean install -Pqulice`. Follow the recommendations and fix static analyzer errors

**Comment by vkuchyn** [comment]
@sLisnychyi Steps to reproduce:
1. clonning you code `git clone https://github.com/sLisnychyi/microservices.git`
2. Checkout branch `git checkout \#1user_actions`
3. Run maven build script ` mvn clean install`

**Comment by sLisnychyi** [comment]
@vkuchyn 
1. fixed compilation error
2. make some changes according code checker

thank you.

**Comment by vkuchyn** [comment]
@sLisnychyi Thanks for fixes, we are almost done! 
Please, checkout to folder gamifaication and run `mvn clean install -Pqulice`. I see 
`[ERROR] Failed to execute goal com.qulice:qulice-maven-plugin:0.16:check (default) on project microservices-gamification: Failure: 357 Checkstyle violations (see log above) -> [Help 1]
`
I know, it looks horror, but it's nice challenge to practice refactoring ;)

**Comment by vkuchyn** [comment]
@sLisnychyi how is it going? We are blocked a little with this commit, can you speed up a little ) Feel free to ask questions

**Comment by sLisnychyi** [comment]
@vkuchyn 
I have looked for errors. But most of them i don't think we should fix.
There's no so cruel code parts.
If you think we should fix them. Please give more specific example of dirty code

**Comment by vkuchyn** [comment]
@sLisnychyi qulice is list of recommendations out of the box - there is no way (only custom implementations of rules which is too expensive) to change it. No we are trying to use it in our practice. Of course, for the first time it will annoy and make problems - (like wash hands for children also annoying but it is useful rule), but after some time it will give us understanding if these rules are ok for us or not. 
BTW, it is best practices developed by Google and other top companies. To agree or disagree with them requires strong arguments. Just relax and fix all the recommendations ;) I am sure, you can improve with this tool your codestyle.

**Comment by sLisnychyi** [comment]
1)
 Name 'UUID' must match pattern '^(id|[a-z]{3,})$'. (ParameterNameCheck)
Name 'mongoTemplate' must match pattern '^(id|[a-z]{3,12})$'. (MemberNameCheck)

i think camelcase and specific names could be used in code. 
By theirs pattern, i see that valid only lowercase variables from 3 to 12 digits

2)
 [ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[56]: Empty line inside method (EmptyLinesCheck)
[ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[57]: 'if' construct must use '{}'s. (NeedBracesCheck)
[ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[58]: 'if' construct must use '{}'s. (NeedBracesCheck)

should we also fix code in automatic generated (hascode/equals) ?

3)
[ERROR] \src\main\java\juja\microservices\gamification\jersey\UserControllerImpl.java[8]: Using the '.*' form of import should be avoided - javax.ws.rs.*. (AvoidStarImportCheck)

why. it's convenient to import all dependencies from package. especially if you are using a lot of classes from there.

4)
[ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[9]: Lines in file should end with Unix-like end of line (RegexpMultilineCheck)

don't understand yet, what does it mean

**Comment by sLisnychyi** [comment]
@vkuchyn pls check

**Comment by vkuchyn** [comment]
@sLisnychyi thank for you questions. Answering:

> 1)
> Name 'UUID' must match pattern '^(id|[a-z]{3,})$'. (ParameterNameCheck)
> Name 'mongoTemplate' must match pattern '^(id|[a-z]{3,12})$'. (MemberNameCheck)
> 
> i think camelcase and specific names could be used in code. 
> By theirs pattern, i see that valid only lowercase variables from 3 to 12 digits

http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

> 2)
> [ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[56]: Empty line inside method (EmptyLinesCheck)
> [ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[57]: 'if' construct must use '{}'s. (NeedBracesCheck)
> [ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[58]: 'if' construct must use '{}'s. (NeedBracesCheck)
> 
> should we also fix code in automatic generated (hascode/equals) ?

Intellij IDEA provides tool to change autogenerated templates. Default templates can break some specific team code conventions that we have. Just improve your template and you won't have these violations in future.

> 3)
> [ERROR] \src\main\java\juja\microservices\gamification\jersey\UserControllerImpl.java[8]: Using the '.' form of import should be avoided - javax.ws.rs.. (AvoidStarImportCheck)
> 
> why. it's convenient to import all dependencies from package. especially if you are using a lot of classes from there.

Read http://checkstyle.sourceforge.net/config_imports.html 
https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

> 4)
> [ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[9]: Lines in file should end with Unix-like end of line (RegexpMultilineCheck)
> 
> don't understand yet, what does it mean

This rule fix cross-platform issue that could happen 
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Comment by vkuchyn** [comment]
@sLisnychyi better a little, but I still see `264 Checkstyle violations (see log above) -> [Help 1]`. Do you see the same?

**Comment by sLisnychyi** [comment]
@vkuchyn 

> [ERROR] \src\main\java\juja\microservices\gamification\model\entity\User.java[9]: Lines in file should end with Unix-like end of line (RegexpMultilineCheck)
don't understand yet, what does it mean
This rule fix cross-platform issue that could happen 
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

by this link i only see problems with files. But what should i do in .java file to fix this issue. Can't find any valid answer. Please give me some example.

Also the following errors that i don't understand yet
1. Missing a header - not enough lines in file. (HeaderCheck)
? what header should i add?

2. UserController.java[0]: Cant find method comment (MultilineJavadocTagsCheck)
? what kind of method comment at the '0' line

3. Problem finding class/interface comment (JavadocTagsCheck)
? As for me we should add documentation only on some specific logic. So, for my point of view adding document on getters/setters, constructor, toString, hashcode/equals is useless.

If not. Could you please provide maybe some tool for documentation autogeneration.
or show me sample of canonical method comment that 'qualice' compiler will agree that.

4. Also there's a lot in error in autogeneratet hashcode/equals
? i think we should fix this, because this code we didn't create.

**Comment by sLisnychyi** [comment]
@vkuchyn 
I've made the most relevant code change according qualice check.
please review.

**Comment by vkuchyn** [comment]
@sLisnychiy thank you. I will check this evening

**Comment by vkuchyn** [comment]
@sLisnychyi answering questions:
1. 
> by this link i only see problems with files. But what should i do in .java file to fix this issue. Can't find any valid answer. Please give me some example.

I suppose, you have Windows (cause, I'm on Mac - no violations with `Unix-like end`). Maybe, you have some settings like end of line \n\r or something like this. Google this problem, cause I cannot reproduce it - so hard to help you, it depends on your local settings

2. 
> 1. Missing a header - not enough lines in file. (HeaderCheck)
? what header should i add?

I don't know why in your PR removed my classes from initial commit, you can find [it here ](https://github.com/JuniorsJava/microservices/tree/master/gamification/src/main/java/juja/microservices/gamification). Look at class Starter - at the top of file, before package line copy of license content - this is a policy. Each java class should start with license.

3. 
> Problem finding class/interface comment (JavadocTagsCheck)
> ? As for me we should add documentation only on some specific logic. So, for my point of view adding document on getters/setters, constructor, toString, hashcode/equals is useless. 
> If not. Could you please provide maybe some tool for documentation autogeneration.
or show me sample of canonical method comment that 'qualice' compiler will agree that.

I agree with you - but this tool cannot be configured in the way you are talking about, we cannot switch off comments only for getters, etc. So, should follow those redundant rules. I don't know tool that generates javadoc for setters, getters, etc. BTW, lots of getters, setters methods means weak design, such classes breaks Immutability principle. So, you need to write docs manually.

4. 

> Also there's a lot in error in autogeneratet hashcode/equals ? i think we should fix this, because this code we didn't create.

I think, developer - code author is responsible for code style, even if he (she) generates code with some magic tool. I've told about this, if template of auto generated code provided by IDEA breaks some rules - that means that template does not satisfies our code conventions, it should be changed, and all the violations in generated code should be fixed.

**Comment by vkuchyn** [comment]
> @vkuchyn 
> I've made the most relevant code change according qualice check.
> please review.

@sLisnychyi Thanks for your job. With small steps we are moving to the final end :) I still see 213 violations. Task is to fix all of them, even if you against some of them.

**Comment by sLisnychyi** [comment]
@vkuchyn 
fixed the most relevant. pls check

**Comment by vkuchyn** [comment]
@sLisnychyi nice job, thanks. Only 94 violations left.
Lot of them we were talking about - I believe, you can fix them easily
`[ERROR] /src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java[0]: Missing package-info.java file. (JavadocPackageCheck)` - add package-info.java for each package
`[ERROR] /src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java[39]: Wrong order for 'java.util.List' import. (ImportOrderCheck)` - imports should be in alphabetical order
`[ERROR] /src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java[43]: Two consecutive empty lines (RegexpMultilineCheck)` means redundant empty line after import block
`[ERROR] /src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java[46]: First sentence should end with a period. (JavadocStyleCheck)` Java doc comment sentence should  ends with dot "."
`[ERROR] /src/main/java/juja/microservices/gamification/jersey/UserControllerImpl.java[77]: Return count is 2 (max allowed is 1). (ReturnCountCheck)` - should be only 1 return statement in method. 
```
[ERROR] /src/main/java/juja/microservices/gamification/model/entity/User.java[119]: Parameter o should be final. (FinalParametersCheck)
[ERROR] /src/main/java/juja/microservices/gamification/model/entity/User.java[119]: Name 'o' must match pattern '^(id|[a-z]{3,})$'. (ParameterNameCheck)
``` var name should contains at least 3 letters, no capital letters (see [this article](http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html) )
`[ERROR] /src/main/java/juja/microservices/gamification/model/repository/UserRepositoryImpl.java[58]: Method 'createUser' is not designed for extension - needs to be abstract, final or empty. (DesignForExtensionCheck)` - I can explain this rule in other words - class should be or abstract or final. 
And so on. Lots of violations are about javadoc, grammar, just follow description to fix them

**Comment by sLisnychyi** [comment]
@vkuchyn fixed

**Comment by vkuchyn** [comment]
@sLisnychyi nice to hear. But I still get 39 violations .. ) Green mile ! )) I cannot merge if even 1 violation. I understand that you are tired, but let's do last push! )

**Comment by sLisnychyi** [comment]
@vkuchyn 
mostly fixed. left:
- package-info.java exceptions
- exceptions with doc as a header of every class
- indentation exceptions

**Comment by vkuchyn** [comment]
@sLisnychyi many thanks for your job and patient! I will finish it by myself

**Pull request merged**